### DISCUSSION TITLE
Tree view field needs a visual indicator if there are children for an item

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#13318] [Feature Requests]
The tree view field is awesome, but there seems to be no way to detect if there are children nodes other than just randomly clicking on an item and seeing. Could there be some visual indicator that there are indeed child nodes.

I'm guessing this is more than visual presentation only because maybe at the time of pulling the root tree data, perhaps it doesn't know if there are child nodes or not?

**Comment by rijkvanzanten** [comment]
> I'm guessing this is more than visual presentation only because maybe at the time of pulling the root tree data, perhaps it doesn't know if there are child nodes or not?

That's the reason why it currently doesn't. The child items are loaded in asynchronously, so the parent only know they exist once it's opened (and therefore fetches the nested items)

**Reply by sxsws** [reply]
so it would need to make a separate query for each item to determine the count of children which will get.. horrible quickly? 

Not sure how it'll relate, but maybe you need to add an option for collections to add the necessary fields (that can be managed by the system) for effective tree management. I last used https://github.com/doctrine-extensions/DoctrineExtensions/blob/main/doc/tree.md to implement the fields for a tree graph.

**Reply by rijkvanzanten** [reply]
> so it would need to make a separate query for each item to determine the count of children which will get.. horrible quickly?

Not necessarily. The API already has ways to get that nested count of records efficiently, the app just doesn't currently use it. We should be able to update the app to fetch both current parent level children + the count of children per child in a single query. The main thing you want to avoid here is fetching the whole tree at once, as you could have a very large number of items / recursion

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
